src/utils.js

```diff
@@ -0,0 +1,13 @@
+import { interpolateSpectral, scaleSequential, scaleLinear } from 'd3';
+
+// Create spectral color generator function
+export function generateSpectralInterpolate(min_value, max_value) {
+  return scaleSequential()
+    .domain([min_value, max_value])
+    .interpolator(interpolateSpectral);
+}
+
+// Create linear color generator function
+export function generateLinearColor(min_value, max_value, color) {
+  return scaleLinear().range(['#ffffff', color]).domain([min_value, max_value]);
+}
```

Comment:
Merge into 1 function. Input args should be `color: string` `min` and `max`.

Use a switch block inside the function that operates over `color` input